{
  Throwable failure=new ContextualException();
  DefaultExceptionAnalyser analyser=analyser();
  Throwable transformedFailure=analyser.transform(failure);
  assertThat(transformedFailure,instanceOf(LocationAwareException.class));
  LocationAwareException gse=(LocationAwareException)transformedFailure;
  assertThat(gse.getCause(),sameInstance(failure));
  assertThat(gse.getReportableCauses(),isEmpty());
}
